desc/coils.py

```diff
@@ -1498,6 +1505,99 @@ def to_SplineXYZ(self, knots=None, grid=None, method="cubic", name=""):
         coils = [coil.to_SplineXYZ(knots, grid, method) for coil in self]
         return self.__class__(*coils, NFP=self.NFP, sym=self.sym, name=name)
 
+    def is_self_intersecting(self, grid=None, tol=None):
+        """Check if any coils in the CoilSet intersect.
+
+        By default, checks intersection by checking that for each point on a given coil
+        the closest point in the coilset is on that same coil. If the closest point is
+        on another coil, that indicates that the coils may be close to intersecting.
+
+        If instead the ``tol`` argument is provided, then the function will
+        check the minimum distance from each coil to each other coil against
+        that tol and if it finds the minimum distance is less than the ``tol``,
+        it will take it as intersecting coils, returning True and raising a warning.
+
+        NOTE: If grid resolution used is too low, this function may fail to return
+        the correct answer.
+
+        Parameters
+        ----------
+        grid : Grid, optional
+            Collocation grid containing the nodes to evaluate the coil positions at.
+            If a list, must have the same structure as the coilset. Defaults to the
+            default grid for each coil type, check the docstrings and ``curve.py``
+            for more details.
+        tol : float, optional
+            the tolerance (in meters) to check the intersections to, if points on any
+            two coils are closer than this tolerance, then the function will return
+            True and a warning will be raised. If not passed, then the method used
+            to determine coilset intersection will be based off of checking that
+            each point on a coil is closest to a point on the same coil, which does
+            not rely on a ``tol`` parameter.
+
+        Returns
+        -------
+        is_self_intersecting : bool
+            Whether or not any coils in the CoilSet come close enough to eachother to
```

Comment:
typo: "each other"
This returned bool doesn't actually get used anywhere? That's fine, just want to make sure I understand what's going on.

Reply:
the returned bool is up to the user, it is like is_nested()

It is called in from_symmetry simply to check if the coils are intersecting